src/s390x/zipl.rs

```diff
@@ -330,73 +330,37 @@ pub fn zipl<P: AsRef<Path>>(
         )
     } else {
         // This branch could be also executed during installation, that's why
-        // we have to take care of ignition.firstboot karg and copy bls config
-        // files for further modification
-        let tempdir = Builder::new()
-            .prefix("coreos-installer-zipl-bls-")
-            .tempdir()
-            .context("creating temporary directory")?;
+        // we have to take care of ignition.firstboot karg
         let firstboot_file = boot.join("ignition.firstboot");
-        let blsdir = if kargs.is_some() || firstboot_file.exists() {
-            let blsdir = tempdir.path().join("loader/entries");
-            create_dir_all(&blsdir).with_context(|| format!("creating {}", blsdir.display()))?;
-            read_dir(boot.join("loader/entries"))
-                .with_context(|| format!("reading {}", boot.display()))?
-                .filter_map(Result::ok)
-                .filter(|p| p.file_type().unwrap().is_file())
-                .for_each(|src| {
-                    copy(src.path(), blsdir.join(src.file_name())).unwrap();
-                });
-
-            let mut extra = Vec::new();
-            if firstboot_file.exists() {
-                extra.push("ignition.firstboot".to_string());
-                let firstboot_contents = std::fs::read_to_string(&firstboot_file)
-                    .with_context(|| format!("reading \"{}\"", firstboot_file.display()))?;
-                if let Some(firstboot_kargs) = extract_firstboot_kargs(&firstboot_contents)? {
-                    extra.extend_from_slice(
-                        &firstboot_kargs
-                            .split_whitespace()
-                            .map(|s| s.to_string())
-                            .collect::<Vec<String>>(),
-                    );
-                }
+        let (kernel, initrd, mut options) = get_info_from_bls(boot)?;
+        // we need a full path to kernel and initrd
```

Comment:
Shouldn't this be e.g.

```suggestion
        // we need a relative path to kernel and initrd
```

?

Reply:
Does this then work because `--image` and `--ramdisk` are looked for relative to `--target`?

Reply:
I'm confused.

On line 337 and 338 below he is making the paths full absolute paths, which matches his comment of `we need a full path to kernel and initrd`.

Reply:
Right yes, I misread those lines.

Reply:
Does that resolve this line of questioning then?